pkg/storageops/vsphere/vsphere.go

```diff
@@ -91,6 +95,59 @@ func (ops *vsphereOps) Create(opts interface{}, labels map[string]string) (inter
 		return nil, err
 	}
 
+	isPod, storagePod, err := IsStoragePod(ctx, vmObj, volumeOptions.Datastore)
```

Comment:
suggestion: You can break the Create function and separate out the datastore selection piece into its own function.

Reply:
done. Separated into a new function `getDatastoreToUseInStoragePod`.